include/dump_offload.hpp

```diff
@@ -0,0 +1,518 @@
+                                  std::shared_ptr<Handler>>
+    systemHandlers;
+
+inline void resetHandlers()
+{
+    if (!systemHandlers.empty())
+    {
+        auto handler = systemHandlers.begin();
+        if (handler == systemHandlers.end())
+        {
+            BMCWEB_LOG_DEBUG << "No handler to cleanup";
+            return;
+        }
+        if ((handler->second->dumpType == "system") ||
+            (handler->second->dumpType == "resource"))
+        {
+            handler->first->close();
+            BMCWEB_LOG_CRITICAL << "INFO: " << handler->second->dumpType
+                                << " dump resetHandlers cleanup";
+        }
+    }
+    return;
+}
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(
+        app,
+        "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/<str>/attachment/")
+        .privileges({{"ConfigureComponents", "ConfigureManager"}})
+        .streamingResponse()
+        .onopen(
+            [](crow::streaming_response::Connection& conn) {
+        if (!bmcHandlers.empty())
+        {
+            BMCWEB_LOG_ERROR << "Can't allow dump offload opertaion, one "
+                                "of the BMC dump is already offloading";
+            conn.sendStreamErrorStatus(
+                boost::beast::http::status::service_unavailable);
+            conn.close();
+            return;
+        }
+
+        std::string url(conn.req.target());
+        std::string startDelimiter = "Entries/";
+        std::size_t pos1 = url.rfind(startDelimiter);
+        std::size_t pos2 = url.rfind("/attachment");
+        if (pos1 == std::string::npos || pos2 == std::string::npos)
+        {
+            BMCWEB_LOG_ERROR << "Unable to extract the dump id";
+            conn.sendStreamErrorStatus(boost::beast::http::status::not_found);
+            conn.close();
+            return;
+        }
+
+        std::string dumpId = url.substr(pos1 + startDelimiter.length(),
+                                        pos2 - pos1 - startDelimiter.length());
+
+        std::string dumpType = "bmc";
+        boost::asio::io_context* ioCon = conn.getIoContext();
+
+        // Generating random id to create unique socket file
+        // for each dump offload request
+        std::random_device rd;
+        std::default_random_engine gen(rd());
+        std::uniform_int_distribution<> dist{0, 1024};
+        std::string unixSocketPath =
+            unixSocketPathDir + dumpType + "_dump_" + std::to_string(dist(gen));
+
+        bmcHandlers[&conn] =
+            std::make_shared<Handler>(*ioCon, dumpId, dumpType, unixSocketPath);
+        bmcHandlers[&conn]->connection = &conn;
+
+        if (!crow::ibm_utils::createDirectory(unixSocketPathDir))
+        {
+            bmcHandlers[&conn]->connection->sendStreamErrorStatus(
+                boost::beast::http::status::not_found);
+            bmcHandlers[&conn]->connection->close();
+            return;
+        }
+        BMCWEB_LOG_CRITICAL
```

Comment:
Really CRITICAL?

Reply:
yeah these are really needed to debug @gtmills 
we have added these traces as per HMC -BMC issues we have seen